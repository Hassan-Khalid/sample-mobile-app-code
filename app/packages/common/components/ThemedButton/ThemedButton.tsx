import React from 'react';
import { Button } from '@rneui/themed';
import styles from "./styles/ThemedButton.styles";
import Util from '../../util/Util';
import { t } from 'i18next';
import componentBuilder from 'common/builder/ComponentBuilder';

const btnSize = {
  large : {width:Util.getWindowWidth() - 50},
  medium : {width:Util.getWindowWidth() - 150},
  small :{width:Util.getWindowWidth() - 250}
  
}

const ThemedButton = (props) => {
  const {size}= props
  const _s = btnSize[size]
  
  return (
    <Button
      {...componentBuilder.createTestAttributes('button-component')}
      title={t(props.title)}
      onPress={ !props.loading && props.onPress}
      disabled={props.disabled}
      type={props.type}
      loading={props.loading}
      buttonStyle={[styles.buttonStyle, props.buttonStyle,_s]}
      titleStyle={[styles.titleStyle, props.titleStyle]}
    />
  );
};

export default ThemedButton
