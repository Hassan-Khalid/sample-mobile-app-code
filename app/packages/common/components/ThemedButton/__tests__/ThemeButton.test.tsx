import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ThemedButton from '../ThemedButton';

// Mocking the i18next t function
jest.mock('i18next', () => ({
  t: jest.fn(text => text), // Mocking t function to return the key as is
}));

jest.mock('common/util/Util');
describe('ThemedButton component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  })
  
  it('renders correctly ', () => {
    // exercise
    const wrapper = render(<ThemedButton/>);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  it('renders button with correct properties', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <ThemedButton
        title="Test Button"
        onPress={onPressMock}
        disabled={false}
        type="solid"
        loading={false}
        size="large"
        buttonStyle={{ backgroundColor: 'red' }}
        titleStyle={{ color: 'blue' }}
      />
    );
    
    // Check if the button is rendered
    const button = getByText('Test Button');
    expect(button).toBeTruthy();
    
    // Simulate button press event
    fireEvent.press(button);
    
    // Check if onPress handler is called
    expect(onPressMock).toHaveBeenCalled();
  });
  
  
  // Add more test cases as needed
});
