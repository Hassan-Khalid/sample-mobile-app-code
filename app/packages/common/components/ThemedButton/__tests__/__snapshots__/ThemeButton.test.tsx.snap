// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedButton component renders correctly  1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "borderRadius": 2,
      },
      undefined,
      false,
    ]
  }
  testID="RNE_BUTTON_WRAPPER"
>
  <View
    accessibilityLabel="button-component"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": false,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button-component"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0000EE",
          "borderColor": "#2089dc",
          "borderRadius": 10,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 30,
          "padding": 8,
          "paddingHorizontal": 10,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
